hera_cal/abscal_funcs.py

```diff
@@ -1055,56 +1063,83 @@ def interp2d_vis(model, model_lsts, model_freqs, data_lsts, data_freqs,
 
     data_freqs : 1D array of the data freq axis, dtype=float, shape=(Nfreqs,)
 
-    kind : kind of interpolation method, type=str, options=['linear', 'cubic', ...]
-        see scipy.interpolate.interp2d for details
+    flags : type=DataContainer, dictionary containing model flags. Can also contain model wgts and
+                                will convert appropriately.
+
+    medfilt_flagged : type=bool, if True, run a median filter on flagged model pixels
 
-    fill_value : values to put for interpolation points outside training set
-        if None, values are extrapolated
+    medfilt_window : type=tuple, window for median filter across the (time, freq) axes.
 
-    zero_tol : for amplitudes lower than this tolerance, set real and imag components to zero
+    kx : type=int, type of spline to fit to freq axis of data. See scipy.interpolate.RectBivariateSpline
 
-    bounds_error : type=boolean, if True, raise ValueError when extrapolating. If False, extrapolate.
+    ky : type=int, type of spline to fit to time axis of data. See scipy.interpolate.RectBivariateSpline
 
-    flag_extrapolate : flag extrapolated data if True
+    s : type=int, smoothing factor for interpolation. Default is no smoothing. See scipy.interpolate.RectBivariateSpline
+
+    flag_extrapolate : type=bool, flag extrapolated data if True
 
     Output: (new_model, new_flags)
     -------
     new_model : interpolated model, type=dictionary
-    new_flags : flags associated with model, type=dictionary
+    new_flags : flags associated with interpolated model, type=dictionary
     """
     # make flags
     new_model = odict()
-    flags = odict()
+    new_flags = odict()
+
+    # get nearest neighbor points
+    freq_nn = np.array(map(lambda x: np.argmin(np.abs(model_freqs-x)), data_freqs))
+    time_nn = np.array(map(lambda x: np.argmin(np.abs(model_lsts-x)), data_lsts))
+    freq_nn, time_nn = np.meshgrid(freq_nn, time_nn)
 
     # loop over keys
     for i, k in enumerate(model.keys()):
-        # interpolate real and imag separately
+        # get model array
         m = model[k]
+
+        # get real and imag separately
         real = np.real(m)
         imag = np.imag(m)
 
-        # interpolate
-        interp_real = interpolate.interp2d(model_freqs, model_lsts, real,
-                                           kind=kind, fill_value=np.nan, bounds_error=bounds_error)(data_freqs, data_lsts)
-        interp_imag = interpolate.interp2d(model_freqs, model_lsts, imag,
-                                           kind=kind, fill_value=np.nan, bounds_error=bounds_error)(data_freqs, data_lsts)
         # set flags
-        f = np.zeros_like(interp_real, dtype=float)
-        if flag_extrapolate:
-            f[np.isnan(interp_real) + np.isnan(interp_imag)] = 1.0
-        interp_real[np.isnan(interp_real)] = fill_value
-        interp_imag[np.isnan(interp_imag)] = fill_value
+        if flags is not None:
+            f = flags[k][time_nn, freq_nn]
```

Comment:
I do not understand this line at all. 

First off, how are you indexing a 2D flag array with two 2D arrays of indices?

Second, it seems to me like your flags are not being propagated correctly. The propagated flags only come from the nearest neighbor, but each datapoint is an interpolation between multiple neighbors. In theory, it should inherit an OR of the flags from both. 

I think Aaron's suggestion was just to define abs model visibilities such that they had no flags, either by combining days (to get rid of time-dependent) or via delay filtering (to get rid of freq-dependent flags).

Reply:
yeah so `f` is an array containing the nearest neighbor (NN) flag for each interpolated point. it should be a 2d ndarray. 

yeah so the flag propagation is a little tricky. one thing i don't think we want is an OR on all NN flags. this would mean that, for example, for bicubic interpolation that has an RFI flag on a single time and freq pixel, that rfi flags would spread from a 1x1 region to a 4x4 region.

i think its more reasonable, but not as aggressive, to only let the RFI flag propagate if its is a nearest neighbor for one of the interpolation points. the effect this has can be mitigated by combining this with the `medfilt_flagged` option, which will downweight the ability of the RFI flag to severely bias the interpolation.